### PR TITLE
[DPE-3938] Preflight check tests

### PR DESCRIPTION
**Pull request by MiaAltieri**
## Issue
No tests for recent preflight check code

## Solution
add tests

### CODE REVIEW

**Review by dmitry-ratushnyy — DISMISSED** [review]

File: tests/unit/test_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charms.mongodb.v0.upgrade.MongoDBConnection")
+    def test_is_cluster_healthy(self, connection):
+        """Test is_cluster_healthy function."""
+
+        def is_shard_mock_call(*args):
+            if args == ("shard",):
+                return True
+            else:
+                return False
```
Comment (Mehdi-Bendriss) [inline tests/unit/test_upgrade.py]:
```suggestion
            return args == ("shard",)
```

File: tests/unit/test_upgrade.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        def is_replication_mock_call(*args):
+            if args == ("replication",):
+                return True
+            else:
+                return False
```
Comment (Mehdi-Bendriss) [inline tests/unit/test_upgrade.py]:
```suggestion
            return args == ("replication",)
```

File: tests/integration/upgrade/test_upgrade.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+@pytest.mark.group(1)
+async def test_preflight_check(ops_test: OpsTest) -> None:
+    """Verifies that the preflight check can run successfully."""
+    app_name = await get_app_name(ops_test)
+    leader_unit = await find_unit(ops_test, leader=True, app_name=app_name)
+    logger.info("Calling pre-upgrade-check")
+    action = await leader_unit.run_action("pre-upgrade-check")
+    await action.wait()
+
```
Comment (Mehdi-Bendriss) [inline tests/integration/upgrade/test_upgrade.py:66]: we should perhaps in a future PR, also test a case where the pre-upgrade-check is supposed to fail

**Review by Mehdi-Bendriss — DISMISSED** [review]
Thanks Mia

File: tests/integration/upgrade/test_upgrade.py (hunk 3 above)
Comment (MiaAltieri) [reply tests/integration/upgrade/test_upgrade.py:46]: yes, I will add a JIRA ticket

**Review by Mehdi-Bendriss — DISMISSED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]

**Review by dmitry-ratushnyy — APPROVED** [review]

**Pull request merged**